### ISSUE TITLE
Reapplying GraphicsDeviceManager.ApplyChanges() while .IsFullScreen = true may cancel Fullscreen

### ISSUE THREAD

**Issue by tarwe** [MonoGame/MonoGame#5407]
When I call graphics.ApplyChanges() while in fullscreen, despite having graphics.IsFullScreen = true, the game reverts back to windowed. This usually happens right when I call .ApplyChanges() again, but may occasionally take more attempts. 

While in this state, the game is effectively windowed, but the .IsFullScreen property is still true. The problem persists until I set .IsFullScreen back to false, apply, set it back to true and apply again. It's especially bothersome when a user tries to change the resolution while in fullscreen, as this action requires an .ApplyChanges() call while already in fullscreen, thus triggering the problem. The only workaround I see for this is to hop to windowed and back, but this looks very ugly.

To replicate, create a new project and add the following to the Game1 class (rewriting the existing functions where necessary):
     
    KeyboardState last_state;

    protected override void Initialize() {
        last_state = Keyboard.GetState();
        base.Initialize();
    }

    protected override void Update(GameTime gameTime) {

        if(Keyboard.GetState().IsKeyDown(Keys.Enter) && !last_state.IsKeyDown(Keys.Enter)) {
            graphics.IsFullScreen = true;
            graphics.ApplyChanges();
        }
        last_state = Keyboard.GetState();

        base.Update(gameTime);
    }

When running the game, pressing Enter the first time will trigger fullscreen as intended. Pressing Enter again (or in some cases, several times) will trigger the problem. The screen is back to windowed and stays there no matter how many times you press Enter.

**Comment by Jjagg** [comment]
What project template? Are you on a recent develop version?

**Comment by tarwe** [comment]
Version 3.5.1.1679. I initially encountered it in an OpenGL project, but also got it to show up in a Windows DirectX project.

**Comment by Jjagg** [comment]
On develop I can reproduce this on neither of the two.

**Comment by Jjagg** [comment]
Cannot reproduce on 3.5.1 DirectX either. Trying 3.5.1 OpenGL isn't as easy because I'd need to install the older release so I won't do that. Since I can't reproduce this, maybe you can test a develop version and see if it's fixed? @tarwe

**Comment by tarwe** [comment]
I've been looking up a develop version, but I couldn't find it. Could you point me in the right direction?

**Comment by tarwe** [comment]
(Nevermind, found it. I'm blind. Give me a moment)

**Comment by tarwe** [comment]
Just tested in 3.6.0.1301. I can no longer reproduce the problem in DirectX, nor a clean OpenGL project. It still persists in my original OpenGL project, but I initially had to download some stuff with NuGet to even run the template. I tried looking up what exactly I downloaded, but I can't find it. That likely screwed things up, though.

Since I only have code in my project so far, I'll just move it to a clean OpenGL project in the dev version. The problem is fixed as far as I can see. Thanks for the help!
